components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/SubscriberRegistrationInterceptor.java

```diff
@@ -50,6 +50,11 @@ public SubscriberRegistrationInterceptor() {
     @Override
     public void handleMessage(Message message) {
         String username = RestApiUtil.getLoggedInUsername();
+        //by-passes the interceptor if user is an annonymous user
+        if (username.equalsIgnoreCase("wso2.anonymous.user")) {
```

Comment:
Shall we use a RESTAPI constant for this?

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/6923